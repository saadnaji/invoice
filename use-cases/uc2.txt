add_type("chocolate")
add_type("ChocoLate")
add_type("water") 
add_product("pepsi", 100)
add_product("water", 1005)
add_product("chocolate", 191000000)
add_product("chocolate", -5)
add_order(<<["nuts", 29], ["bolts", 31], ["pepsi", 300]>>)
add_order(<<["waTer", 22], ["chOcoLate", 5], ["chOcoLate", 5]>>)
add_order(<<["waTer", 22], ["chOcoLate", -5]>>)
add_type("chips")
add_product("chips", 3000)
add_order(<<["chips", 100]>>)
invoice(1)
invoice(1)
cancel_order(10)
invoice(10)
add_type("")
add_order(<<["chips", 100000000]>>)

