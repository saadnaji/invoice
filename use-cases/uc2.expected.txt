  report:      ok
  id:          0
  products:   
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("chocolate")
  report:      ok
  id:          0
  products:    chocolate
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("ChocoLate")
  report:      product type already in database
  id:          0
  products:    chocolate
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("water")
  report:      ok
  id:          0
  products:    chocolate,water
  stock:       
  orders:      
  carts:       
  order_state: 
->add_product("pepsi",100)
  report:      product not in database
  id:          0
  products:    chocolate,water
  stock:       
  orders:      
  carts:       
  order_state: 
->add_product("water",1005)
  report:      ok
  id:          0
  products:    chocolate,water
  stock:       water->1005
  orders:      
  carts:       
  order_state: 
->add_product("chocolate",191000000)
  report:      ok
  id:          0
  products:    chocolate,water
  stock:       chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_product("chocolate",-5)
  report:      quantity added must be positive
  id:          0
  products:    chocolate,water
  stock:       chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["nuts", 29], ["bolts", 31], ["pepsi", 300]>>)
  report:      some products in order not valid
  id:          0
  products:    chocolate,water
  stock:       chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["waTer", 22], ["chOcoLate", 5], ["chOcoLate", 5]>>)
  report:      duplicate products in order array
  id:          0
  products:    chocolate,water
  stock:       chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["waTer", 22], ["chOcoLate", -5]>>)
  report:      quantity added must be positive
  id:          0
  products:    chocolate,water
  stock:       chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_type("chips")
  report:      ok
  id:          0
  products:    chips,chocolate,water
  stock:       chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_product("chips",3000)
  report:      ok
  id:          0
  products:    chips,chocolate,water
  stock:       chips->3000,chocolate->191000000,water->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["chips", 100]>>)
  report:      ok
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->pending
->invoice(1)
  report:      ok
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->invoiced
->invoice(1)
  report:      order already invoiced
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->invoiced
->cancel_order(10)
  report:      order id is not valid
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->invoiced
->invoice(10)
  report:      order id is not valid
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->invoiced
->add_type("")
  report:      product type must be non-empty string
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->invoiced
->add_order(<<["chips", 100000000]>>)
  report:      not enough in stock
  id:          1
  products:    chips,chocolate,water
  stock:       chips->2900,chocolate->191000000,water->1005
  orders:      1
  carts:       1: chips->100
  order_state: 1->invoiced
